.github/workflows/stackhpc-all-in-one.yml

```diff
@@ -77,6 +77,10 @@ jobs:
       KAYOBE_VAULT_PASSWORD: ${{ secrets.KAYOBE_VAULT_PASSWORD }}
       KAYOBE_IMAGE: ${{ inputs.kayobe_image }}
     steps:
+      - name: Checkout submodules
```

Comment:
Isn't this done in the following step?

Reply:
This is to try and [workaround](https://github.com/actions/checkout/issues/758) the error below:

> Error: Input 'submodules' not supported when falling back to download using the GitHub REST API

Seemingly, when git tries to tarball git it uses an old one which doesn't work with submodules.

Reply:
can you try checkout/v4 and submodules: recursive?
https://github.com/actions/checkout

Reply:
Will try checkoutv4